projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewStoreReader.java

```diff
@@ -445,10 +441,28 @@ List<Map<String, Set<ValueDTO>>> retrieveJoinTableRows(
         return rows;
     }
 
+    private PreparedStatement getQuery(String view, View.JoinView joinView, Table joinedTable, Table joinTable,
+            List<String> projectionColumns, List<String> ids) throws SQLException {
+
+
+        Function<String, String> queryBody = id
+                -> "(select " + projectionColumns.stream().map(String::toLowerCase).collect(Collectors.joining(", ")) + " , '" + id + "' AS rowid"
+                + " from " + joinedTable.name + " j "
+                + " where exists ("
```

Comment:
StringBuilder is more performant on many string appendings and requres less memory as '+' produce a new string due to immutable nature of string. Sometimes, compiler optimaize it and use stringbuilder under the hood, but not sure if it's been used here. I would explicitly use StringBuilder.

On the other hand, StringBuilder is less concise, but still I think it makes sense to go with it